modules/apps/wiki/wiki-web/src/main/java/com/liferay/wiki/web/internal/change/tracking/spi/display/WikiPageCTDisplayRenderer.java

```diff
@@ -71,6 +99,96 @@ public String getTitle(Locale locale, WikiPage wikiPage) {
 			wikiPage.getTitle(), " (", wikiPage.getVersion(), ")");
 	}
 
+	@Override
+	public String renderPreview(DisplayContext<WikiPage> displayContext)
+		throws Exception {
+
+//		WikiPage wikiPage = displayContext.getModel();
+
+//
+//		PortletRequest portletRequest =
+//			(PortletRequest)httpServletRequest.getAttribute(
+//				JavaConstants.JAVAX_PORTLET_REQUEST);
+//		PortletResponse portletResponse =
+//			(PortletResponse)httpServletRequest.getAttribute(
+//				JavaConstants.JAVAX_PORTLET_RESPONSE);
+//
+//		PortletRequestModel portletRequestModel = new PortletRequestModel(
+//			portletRequest, portletResponse);
+//
+//		String attachmentURLPrefix = StringBundler.concat(
+//			themeDisplay.getPathMain(), "/wiki/get_page_attachment?p_l_id=",
+//			themeDisplay.getPlid(), "&nodeId=", wikiPage.getNodeId(), "&title=",
+//			URLCodec.encodeURL(wikiPage.getTitle()), "&fileName=");
+//
+//		String url = null;
+//
+//		String urlw = _wikiEngineRenderer.convert(
+//			displayContext.getModel(), null, null, attachmentURLPrefix);
+//
+//
+//		WikiPageDisplay wikiPageDisplay =
+//			_wikiPageLocalService.getPageDisplay(
+//				wikiPage,  null,
+//				null, attachmentURLPrefix);
+
+		HttpServletRequest httpServletRequest =
+			displayContext.getHttpServletRequest();
+
+		RenderRequest renderRequest =
+			(RenderRequest)httpServletRequest.getAttribute(
+				JavaConstants.JAVAX_PORTLET_REQUEST);
+		RenderResponse renderResponse =
+			(RenderResponse)httpServletRequest.getAttribute(
+				JavaConstants.JAVAX_PORTLET_RESPONSE);
+
+		WikiPage wikiPage = (WikiPage)httpServletRequest.getAttribute(WikiWebKeys.WIKI_PAGE);
+		WikiEngineRenderer wikiEngineRenderer = (WikiEngineRenderer)httpServletRequest.getAttribute(
```

Comment:
No need to get the wikiEngineRenderer from the request. It's already being injected as `_wikiEngineRenderer`